src/framework/MOM_io.F90

```diff
@@ -332,13 +333,27 @@ subroutine create_MOM_file(IO_handle, filename, vars, novars, fields, &
     IsgB = dG%IsgB ; IegB = dG%IegB ; JsgB = dG%JsgB ; JegB = dG%JegB
   endif
 
-  if (domain_set .and. (num_PEs() == 1)) thread = SINGLE_FILE
```

Comment:
Eliminating this line will change the behavior when there is a single PE in use but `PARALLEL_RESTARTFILES = True`.  I do not believe that any of our test cases explore this (admittedly strange) combination in any of our regression or TC tests.
Have we done any testing to make sure that the behavior is correct in this case? 

  I think that the safest course of action is simply to restore this line. Alternatively to handling this case explicitly here, we could consider adding error handling in the 5 places in the MOM6 and SIS2 code where PARALLEL_RESTARTFILES is read in to issue a fatal or warning error message if this combination of `PARALLEL_RESTARTFILES=True` but there is a single PE.

Reply:
This line was removed because the FMS2 I/O driver was rewritten to only work for files with a `Domain`.  The value of `thread` sets the value of `one_file` below, which was determining whether a `Domain` was served to `IO_handle%open()`.

(The reason for this was the overall volatility of generic single-file FMS2 I/O for certain combinations of compilers and files.  Since many of our inputs do not follow FMS conventions, we could never expect them to be handled safely.  Hence, we only divert files with domains to the FMS2 layer.)

I could imagine supporting the FMS2 domainless single-file I/O someday, but more work would need to happen on the FMS2 side for that to happen.

----

As for the scenario you describe:

Restart files always pass `G` and `GV`, so they are always written with a domain.  `PARALLEL_RESTARTFILES = True` determines whether `threading` is set to `SINGLE_FILE` or `MULTIPLE`.  When set to `SINGLE_FILE` (your scenario), a dummy [1,1] IO_domain is created here:

https://github.com/NOAA-GFDL/MOM6/blob/417c0b6542d8b141eb9194d46d0a215371f89468/src/framework/MOM_io_file.F90#L931-L944

I believe I must have tested for this scenario, because this whole implementation is kind of crazy.  But I believe that it is necessary to handle files from SIS2 or other output outside of MOM6.  (In other words, I *think* that I did test the case you mentioned, though I'll need to go back and confirm.)

----

I fully admit there is a lot of oddity to this solution, so we should talk it over next week, but at the least there some rationale behind it.

(BTW: At the very least, the comments hanging around that line of code should have been deleted.  I also think the overall logic in these blocks is a bit wild and could use a rewrite anyway.)

Reply:
FWIW, I tested `PARALLEL_RESTARTFILES = True` on single-core in a few tests and it seemed to work OK.